File: src/main/java/com/exsoloscript/mysteryteams/util/Colors.java
package com.exsoloscript.mysteryteams.util;

import com.exsoloscript.mysteryteams.team.MysteryTeam;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Colors {
    private List<MysteryTeam.ColorData> availableColors;

    public Colors() {
        reset();
    }

    public MysteryTeam.ColorData getNext() {
        if (availableColors.size() > 0) {
            return availableColors.remove(0);
        }

        return null;
    }

    public void reset() {
        this.availableColors = new ArrayList<>(Arrays.asList(MysteryTeam.ColorData.values()));
    }

}

File: src/main/kotlin/com/exsoloscript/mysteryteams/util/Colors.kt
package com.exsoloscript.mysteryteams.util

import com.exsoloscript.mysteryteams.team.MysteryTeam

import java.util.ArrayList
import java.util.Arrays

class Colors {
    private var availableColors: MutableList<MysteryTeam.ColorData>? = null

    init {
        reset()
    }

    val next: MysteryTeam.ColorData?
        get() {
            if (availableColors!!.size > 0) {
                return availableColors!!.removeAt(0)
            }

            return null
        }

    fun reset() {
        this.availableColors = ArrayList(Arrays.asList(*MysteryTeam.ColorData.values()))
    }

}
